content/300-guides/200-deployment/110-deployment-guides/600-deploying-migrations-from-a-local-environment.mdx

```diff
@@ -0,0 +1,51 @@
+---
+title: 'Deploying migrations from a local environment'
+metaTitle: 'Deploying migrations from a local environment'
+metaDescription: 'Learn how to deploy Node.js and TypeScript applications that are using Prisma Client locally.'
+tocDepth: 2
+---
+
+<TopBlock>
+
+There are instances whereby you might want to run a deployment migration locally. This page outlines some examples of how you can do that and **why its generally a bad practice**.
+
+</TopBlock>
+
+## Local CI/CD pipeline
+
+You want to control the CI/CD process locally. You have a database with 3 environments, development, staging and production. You keep 3 different connection strings which each relate to a different environment as `.env` variables.
+
+Having made some changes to your schema you want to run a migration against your development environment so that those changes are reflected in your database. You run `prisma migrate dev` and the migration is applied.
+
+Next you want to manually sync your staging and production environments too.
+
+### A potential solution
+
+> **_We stongly discourage this solution due to the following reasons, use it at your own discretion!_ **
+>
+> - You risk exposing your production database connection string to version control.
+> - You may accidentally use your production connection string instead and in turn **override or delete your production database**.
+
+You can comment out your development and staging connection strings and run `prisma migrate deploy`, this will update your production database with your new schema changes.
+
+> **Note**: `prisma migrate deploy` should generally be part of an automated CI/CD pipeline, and we do not recommend running this command locally to deploy changes to a production database. It is not generally considered good practice to store the production database URL locally.
+
+### An alternative approach
+
+An alternative approach would be to let an automated CI/CD pipeline handle your staging and production environments.
+
+If your goal is to prototype your schema and still have 3 environments that mock a real world use case then you might consider using the [`db push`](../../../../concepts/components/prisma-migrate/db-push) command.
+
+## Add migrations to a pre-existing database locally.
```

Comment:
This section looks like a shorter version of https://deploy-preview-1537--prisma2-docs.netlify.app/docs/guides/database/developing-with-prisma-migrate/add-prisma-migrate-to-a-project

As far as I understand, it's specifically baselining production that one might perform from local --> production. Is that right, @2color? Is there any other way to do it - a one-off script that you run from your CI/CD setup somehow? 🤷 

I would suggest keeping this section part very focused on this: you can do the baselining tasks from local.

* Change title to "Baselining a production database"
* Very short intro - 'When you add Prisma Migrate to an **existing database**, you must baseline the production database (add link). Baselining is performed **once**, and can be done from a local instance: [possible diagram here of laptop -> list of commands run as part of baselining -> production database]

Reply:
@mhwelander Are you suggesting removing all the stuff i had written and keeping it short and with a diagram to illustrate?

Reply:
@molebox That's right, I would really condense it down to only the point about deploying from local